zummon/src/commonMain/kotlin/companysurvey/TimeSeries.kt

```diff
@@ -48,16 +48,37 @@ data class TimeSeries (
         }
     }
 
-    fun hasFullYear(year: Int): Boolean =
-        this.start <=  Instant.parse("$year-01-01T00:00:00+01:00")
-                && this.calculateEnd() >= Instant.parse("${year+1}-01-01T00:00:00+01:00")
+    fun hasFullYear(year: Int? = null): Boolean {
+        var workingYear = year ?: this.start.toLocalDateTime(TimeZone.of("Europe/Amsterdam")).year
+        return this.start <=  Instant.parse("$workingYear-01-01T00:00:00+01:00")
+                && this.calculateEnd() >= Instant.parse("${workingYear+1}-01-01T00:00:00+01:00")
+    }
+
+    fun lengthEmptyGapsData(): Int {
+        var maxNullSequence = 0
+        var currentNullSequence = 0
+
+        for (value in values) {
+            if (value == null) {
```

Comment:
I think we should check for 0 and NaN.

(I'm pretty sure the ER people will make sure there is never a null or NaN)